cpp/dolfinx/fem/DirichletBC.h

```diff
@@ -168,6 +170,59 @@ class DirichletBC
     _dofs1_g = _dofs0;
   }
 
+  /// Create a representation of a Dirichlet boundary condition with a constant
+  /// value.
+  ///
+  /// @param[in] V The function space to constrain
+  /// @param[in] g The constant value
+  /// @param[in] dofs Degree-of-freedom block indices (@p
+  /// std::vector<std::int32_t>) from the input function space to constrain
+  /// @note The dofs will be unrolled with the dofmap block size
+  template <typename U>
+  DirichletBC(const std::shared_ptr<const fem::Constant<T>>& g, U&& dofs,
+              const std::shared_ptr<const fem::FunctionSpace>& V)
+      : _function_space(V), _g(g), _dofs0(std::forward<U>(dofs))
+  {
+    assert(_function_space);
+    std::visit(
```

Comment:
I don't understand what this is for. The Function interface specifies the type for `g`, so why is there a visitor pattern to check?